indico/modules/rb_new/client/js/common/rooms/RoomPhoto.jsx

```diff
@@ -0,0 +1,171 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2019 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import roomPhotoURL from 'indico-url:rooms_new.admin_room_photo';
+
+import React, {useCallback, useState, useEffect, createRef} from 'react';
+import PropTypes from 'prop-types';
+import Dropzone from 'react-dropzone';
+import {Button, Dimmer, Icon, Image, Popup, Loader} from 'semantic-ui-react';
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+import {Translate} from 'indico/react/i18n';
+
+import './RoomPhoto.module.scss';
+
+
+export default function RoomPhoto({roomId, hasPhoto}) {
+    const [file, setFile] = useState(null);
+    const [loading, setLoading] = useState(false);
+    const [canSave, setCanSave] = useState(false);
+    const [hasPreview, setHasPreview] = useState(false);
+
+    useEffect(() => {
+        setHasPreview(hasPhoto);
+    }, [hasPhoto]);
+
+    const onDrop = useCallback(acceptedFiles => {
+        setFile(acceptedFiles[0]);
+        setHasPreview(true);
+        setCanSave(true);
+    }, []);
+
+    const deletePhoto = useCallback(async () => {
+        setLoading(true);
+        try {
+            await indicoAxios.delete(roomPhotoURL({room_id: roomId}));
+        } catch (e) {
+            handleAxiosError(e);
+        }
+        setFile(null);
+        setLoading(false);
+        setCanSave(false);
+        setHasPreview(false);
+    }, [roomId]);
+
+    const savePhoto = useCallback(async () => {
```

Comment:
are you sure you need `useCallback` here? From what I can see there's no need to force this function's identity to remain the same between renders.

Reply:
I was experimenting mostly and wasn't so sure about this. I guess the same applies to `deletePhoto` as well, right?